### PR TITLE
Compile the `tendermint` and `light-client` crates to WASM

### PR DESCRIPTION
**Pull request by romac**
See: https://github.com/informalsystems/tendermint-rs/issues/463

This PR also adds a job to the CI to test that both crates can compile to WASM.

* [x] Referenced an issue explaining the need for the change
* [ ] Updated all relevant documentation in docs
* [ ] Updated all code comments where relevant
* [ ] Wrote tests
* [ ] Updated CHANGELOG.md

### CODE REVIEW

**Comment by romac** [comment]
I had to disable the only previously enabled acceptance test of the light node CLI for the test suite to pass (). Without this commit, `cargo test --all-features --no-fail-fast` would fail with:

```
       Doc-tests tendermint-light-client
    error[E0460]: found possibly newer version of crate `tendermint` which `tendermint_rpc` depends on
     --> /Users/coromac/Informal/Code/Current/tendermint-rs/light-client/src/components/io.rs:7:5
      |
    7 | use tendermint_rpc as rpc;
      |     ^^^^^^^^^^^^^^
      |
      = note: perhaps that crate needs to be recompiled?
      = note: the following crate versions were found:
              crate `tendermint`: /Users/coromac/Informal/Code/Current/tendermint-rs/target/debug/deps/libtendermint.rlib
              crate `tendermint`: /Users/coromac/Informal/Code/Current/tendermint-rs/target/debug/deps/libtendermint-735716e8d731096d.rmeta
              crate `tendermint_rpc`: /Users/coromac/Informal/Code/Current/tendermint-rs/target/debug/deps/libtendermint_rpc-ad41c355acf2752f.rlib

    error: aborting due to previous error
```

It is currently unclear why this error has appeared, though I suspect it is a bug in Cargo related to features flags (which we now use to disable non WASM-compatible code).

**Comment by romac** [comment]
Since the test I had to comment out was just testing that the version number emitted by the light node CLI was correct, I feel like this should not be a blocker to merge this PR, though we will at some point need to figure out how to fix the issue anyhow (or just hope that a Cargo update will resolve it, might open an issue on the Rust bugtracker or post on /r/reddit in the meantime).

**Comment by romac** [comment]
Light Client Browser

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -157,3 +157,17 @@ jobs:
...
+      - uses: actions/checkout@v2
+      - uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          override: true
+          target: wasm32-unknown-unknown
+      - name: Build Tendermint for WASM
+        run: cd tendermint && cargo build --target wasm32-unknown-unknown --release --no-default-features
+      - name: Build Light Client for WASM
+        run: cd light-client && cargo build --target wasm32-unknown-unknown --release --no-default-features
```
Comment (liamsi) [inline .github/workflows/rust.yml:173]: Wohoooo 🎉

**Review by melekes — APPROVED** [review]
👍

**Review by brapse — APPROVED** [review]
Love this 👍

**Review by brapse — APPROVED** [review]
👍

**Pull request merged**